Sources/PackageCollections/Providers/JSONPackageCollectionProvider.swift

```diff
@@ -0,0 +1,364 @@
+        let packageName: String
+        let toolsVersion: String
+        let targets: [Target]
+        let products: [Product]
+        let minimumPlatformVersions: [PlatformVersion]?
+        let verifiedPlatforms: [Platform]?
+        let verifiedSwiftVersions: [String]?
+        let license: License?
+
+        enum CodingKeys: String, CodingKey {
+            case version
+            case packageName
+            case toolsVersion
+            case targets
+            case products
+            case minimumPlatformVersions
+            case verifiedPlatforms
+            case verifiedSwiftVersions
+            case license
+        }
+    }
+
+    fileprivate struct Target: Decodable {
+        let name: String
+        let moduleName: String?
+    }
+
+    fileprivate struct Product: Decodable {
+        let name: String
+        let type: ProductType
+        let targets: [String]
+
+        enum CodingKeys: String, CodingKey {
+            case name
+            case type
+            case targets
+        }
+    }
+
+    fileprivate enum ProductType: Decodable {
+        case library(LibraryType)
+        case executable
+        case test
+
+        enum LibraryType: String, Decodable {
+            case `static`
+            case dynamic
+            case automatic
+        }
+
+        public enum CodingKeys: CodingKey {
+            case library
+            case executable
+            case test
+        }
+
+        init(from decoder: Decoder) throws {
+            let container = try decoder.container(keyedBy: CodingKeys.self)
+            if container.contains(.library) {
+                let types = try container.decode([LibraryType].self, forKey: .library)
+                // FIXME: format defines as array, but model does not
+                guard let type = types.first else {
+                    throw Errors.invalidLibraryType
+                }
+                self = .library(type)
+            } else if container.contains(.executable) {
+                self = .executable
+            } else if container.contains(.test) {
+                self = .test
+            } else {
+                throw Errors.unknownProductType
+            }
+        }
+
+        enum Errors: Error {
+            case invalidLibraryType
+            case unknownProductType
+        }
+    }
+
+    fileprivate struct Platform: Decodable {
+        let name: String
+    }
+
+    fileprivate struct PlatformVersion: Decodable {
+        let name: Platform
+        let version: String
+    }
+
+    fileprivate struct License: Decodable {
+        let name: String
+        let url: String
+    }
+
+    fileprivate struct Author: Decodable {
+        let name: String
+    }
+}
+
+// MARK: - Extensions for mapping from JSON to PackageCollectionsModel
+
+extension PackageCollectionsModel.PackageProduct {
+    fileprivate init?(from: JSONPackageCollectionProvider.CollectionV1.Product, packageTargets: [PackageCollectionsModel.PackageTarget]) {
```

Comment:
Either we don't need this or we can simplify this. 
- `ProductType` shouldn't require special processing. 
- Not sure why we need to do `packageTargets.filter` here.

Reply:
still need to filter tarvgterts since we are going from array of strings to actual targets, the filter is on the packageTargets to scope them down to the product targets

Reply:
> the filter is on the packageTargets to scope them down to the product targets

We seem to be replicating logic but I suppose the JSON can be generated in various ways so we should have it here.

Reply:
yes, we can also address when we consolidate Collection.Package and Package